homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -187,9 +189,23 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 
     # Create handler
     _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
-    vacuum = Vacuum(host, token)
 
-    mirobo = MiroboVacuum(name, vacuum)
+    unique_id = None
+
+    try:
+        vacuum = Vacuum(host, token)
+        device_info = await hass.async_add_executor_job(vacuum.info)
+        unique_id = device_info.mac_address
+        _LOGGER.info(
```

Comment:
Is this a necessary info log message in addition to line 191? You could change it to debug:
```suggestion
        _LOGGER.debug(
```